api/tasks.go

```diff
@@ -1032,14 +1032,17 @@ type TaskCSIPluginConfig struct {
 	// CSIPluginType instructs Nomad on how to handle processing a plugin
 	Type CSIPluginType `mapstructure:"type" hcl:"type,optional"`
 
-	// MountDir is the destination that nomad should mount in its CSI
-	// directory for the plugin. It will then expect a file called CSISocketName
-	// to be created by the plugin, and will provide references into
-	// "MountDir/CSIIntermediaryDirname/VolumeName/AllocID for mounts.
-	//
-	// Default is /csi.
+	// MountDir is the directory (within its container) in which the plugin creates a
+	// socket (called CSISocketName) for communication with Nomad. Default is /csi.
 	MountDir string `mapstructure:"mount_dir" hcl:"mount_dir,optional"`
 
+	// StagePublishDir is the base directory (within its container) in which the plugin
+	// mounts volumes being staged and bind mounts volumes being published.
+	// e.g. staging_target_path = {StagePublishDir}/staging/{volume-id}/{usage-mode}
+	// e.g. target_path = {StagePublishDir}/per-alloc/{alloc-id}/{volume-id}/{usage-mode}
+	// Default is /local/csi.
+	StagePublishDir string `mapstructure:"stage_publish_dir" hcl:"stage_publish_dir,optional"`
```

Comment:
This naming may or may not be controversial (as a new part of Nomad's public API). I tried to make it as descriptive as possible, but I'm definitely open to other ideas.